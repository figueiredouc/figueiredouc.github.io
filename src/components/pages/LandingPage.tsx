import React from 'react';
import 'app/App.css';
import {
  createStyles,
  makeStyles,
  Theme,
  Typography,
  Divider,
  CircularProgress,
} from '@material-ui/core';
import AppPage from 'components/shared/AppPage';
import ProductsList from 'components/shared/ProductsList';
import { useProducts } from 'hooks';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    content: {
      padding: theme.spacing(3),
    },
    divider: {
      backgroundColor: '#B80C09',
    },
    label: {
      color: '#033860',
    },
    loading: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      paddingTop: theme.spacing(3),
    },
  })
);

const LandingPage = () => {
  const classes = useStyles();

  const { productsWithShopCart, isLoadingProducts } = useProducts({
    featured: true,
  });

  return (
    <AppPage showCarousel>
      {isLoadingProducts ? (
        <div className={classes.loading}>
          <CircularProgress />
        </div>
      ) : (
        <div className={classes.content}>
          <Typography className={classes.label} variant="h4">
            Destaques
          </Typography>
          <Divider className={classes.divider} />
          <ProductsList featured products={productsWithShopCart} />
        </div>
      )}
    </AppPage>
  );
};

export default LandingPage;
