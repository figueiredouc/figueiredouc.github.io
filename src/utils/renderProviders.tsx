/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom'; // 4.0.0
import { render } from '@testing-library/react';
import { AuthContext, AuthStore } from 'context/AuthContext';
import {
  BreadcrumbsContext,
  BreadcrumbsStore,
} from 'context/BreadcrumbsContext';
import { ShopCartContext, ShopCartStore } from 'context/ShopCartContext';
import {
  linkedBreadcrumbs as defaultLinkedB,
  activeBreadcrumb as defaultActiveB,
} from '__mocks__/breadcrumbs';
import { shopCart as defaultShopCart } from '__mocks__/products';

export const defaultStore = {
  AuthStore,
  ShopCartStore,
  BreadcrumbsStore,
};

const customRender = (ui: any, { store = defaultStore, ...options } = {}) => {
  const CombinedProviders = ({ children }: any) => {
    return (
      <AuthContext.Provider value={store.AuthStore()}>
        <ShopCartContext.Provider value={store.ShopCartStore()}>
          <BreadcrumbsContext.Provider value={store.BreadcrumbsStore()}>
            <MemoryRouter>{children}</MemoryRouter>
          </BreadcrumbsContext.Provider>
        </ShopCartContext.Provider>
      </AuthContext.Provider>
    );
  };

  return render(ui, { wrapper: CombinedProviders, ...options });
};

export const customStore = ({
  isLoadingAuthStore = false,
  getLoggedUser = jest.fn(),
  createUser = jest.fn(),
  loggedUser = { user: 'xpto', id: 1, token: '123' },
  shopCart = defaultShopCart,
  shopCartTotalPrices = {
    totalPrice: '20',
    totalPriceBeforeTax: '40',
  },
  linkedBreadcrumbs = defaultLinkedB,
  activeBreadcrumb = defaultActiveB,
  useDefaultAuthStore = false,
  useDefaultShopCartStore = false,
  useDefaultBreadcrumbsStore = false,
} = {}) => {
  const newAuthStore = () => ({
    isLoadingUser: isLoadingAuthStore,
    getLoggedUser,
    loggedUser,
    createUser,
  });

  const newShopCartStore = () => ({
    shopCart,
    shopCartTotalPrices,
  });

  const newBreadcrumbsStore = () => ({
    linkedBreadcrumbs,
    activeBreadcrumb,
  });

  return {
    ...defaultStore,
    AuthStore: useDefaultAuthStore ? defaultStore.AuthStore : newAuthStore,
    ShopCartStore: useDefaultShopCartStore
      ? defaultStore.ShopCartStore
      : newShopCartStore,
    BreadcrumbsStore: useDefaultBreadcrumbsStore
      ? defaultStore.BreadcrumbsStore
      : newBreadcrumbsStore,
  };
};

export * from '@testing-library/react';
export { customRender as render };
