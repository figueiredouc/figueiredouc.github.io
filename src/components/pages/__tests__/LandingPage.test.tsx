import React from 'react';
import { render } from 'utils/renderProviders';
import LandingPage from 'components/pages/LandingPage';

describe('renders LandingPage', () => {
  it('should match snapshot', () => {
    const { container } = render(<LandingPage />);

    expect(container).toMatchSnapshot();
  });
});
