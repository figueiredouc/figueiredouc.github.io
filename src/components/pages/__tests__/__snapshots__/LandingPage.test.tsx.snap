// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LandingPage should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-8"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-10 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-header-11 MuiToolbar-gutters"
      >
        <div
          class="makeStyles-logoContainer-12"
        >
          <button
            aria-controls="long-menu"
            aria-haspopup="true"
            aria-label="more"
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <img
            alt="cmai"
            height="50px"
            src="logo.png"
          />
        </div>
        <div
          class="makeStyles-flex-18"
        >
          <div
            class="makeStyles-search-20"
          >
            <div
              class="makeStyles-searchIcon-21"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root makeStyles-inputRoot-22"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input makeStyles-inputInput-23"
                placeholder="Search…"
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
              </span>
              Iniciar Sessão
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
  <div
    class="makeStyles-dividerMargin-14"
  >
    <hr
      class="MuiDivider-root makeStyles-divider-13"
    />
  </div>
  <div
    class="Carousel-root-40 "
  >
    <div
      class="CarouselItem"
    >
      <div
        style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiPaper-root makeStyles-carouselItemContainer-52 MuiPaper-elevation0 MuiPaper-rounded"
        >
          <img
            alt="Random Name #1"
            class="makeStyles-carouselItemImg-53"
            src="https://media.istockphoto.com/vectors/mega-sale-discount-vector-layout-concept-illustration-abstract-vector-id1042356592"
          />
        </div>
      </div>
    </div>
    <div
      class="Carousel-buttonWrapper-44 Carousel-fullHeightHoverWrapper-45 Carousel-next-50"
    >
      <button
        aria-label="Next"
        class="MuiButtonBase-root MuiIconButton-root Carousel-button-46 Carousel-buttonVisible-48 Carousel-fullHeightHoverButton-47 Carousel-next-50"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="Carousel-buttonWrapper-44 Carousel-fullHeightHoverWrapper-45 Carousel-prev-51"
    >
      <button
        aria-label="Previous"
        class="MuiButtonBase-root MuiIconButton-root Carousel-button-46 Carousel-buttonVisible-48 Carousel-fullHeightHoverButton-47  Carousel-prev-51"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiContainer-root makeStyles-container-5 MuiContainer-maxWidthLg"
  >
    <div
      class="makeStyles-loading-4"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
